File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answerlist = []
    newel= elements*2
    
    #2개
    for i in range(1,len(elements)) :
        for j in range(0,len(elements)):
            answerlist.append(sum(newel[j:j+i]))
    answerlist.append(sum(elements))
            
    answer = len(set(answerlist))
    
    return answer

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*; 

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        int [] circle = new int [elements.length*2];
       
        for (int i = 0 ; i<elements.length; i++)
        {
            circle[i] = elements[i];
            circle[elements.length+i] = elements[i];
        }
        
        Set<Integer> answerset = new HashSet<>();

        for (int i = 0; i<elements.length; i++)
        {
            int cnt =  0;
            for (int j = 0; j< elements.length; j++)
            {
                cnt += circle[i+j];
                answerset.add(cnt);
            }
        }
        
        
        return answerset.size();
    }
}